src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -111,6 +111,13 @@ public override BoundNode VisitUserDefinedConditionalLogicalOperator(BoundUserDe
 
         public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryOperator? applyParentUnaryOperator)
         {
+            if (node.InterpolatedStringHandlerData is InterpolatedStringHandlerData data)
```

Comment:
Can we assert that `IsUnconvertedInterpolatedStringAddition` is false? (unconverted nodes don't survive initial binding)

Reply:
We do. See the change in `BoundExpression.cs` to `NeedsToBeConverted`.

Reply:
Yeah, I think the problem is more general than what I thought. We should assert that when we get to lowering all nodes have `NeedsToBeConverted` as false. This is not something specific to this PR.